### PR TITLE
Paginated updateMany

### LINKED ISSUE #937: Support pagination to `UpdateMany` command
Support pagination to `UpdateMany` command. Following are the changes:
- The command will return a `nextPagingState` in case read operation done internally for this command has one. 
- The command will accept a `pagingState` which is used along with the find operation.
- Continue to support `moreData` flag, for backward compatibility but this will be just `nextPagingState != null` and may return a false positive.

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Support pagination in updateMany command
**Which issue(s) this PR fixes**:
Fixes #937 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by vkarpov15** [comment]
Can we add an option to JSON API for the JSON API layer to handle pagination in updateMany? I suppose we can handle this in the client layer as well, but it would be cleaner if JSON API could handle that.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/UpdateManyCommand.java [hunk 1]
```diff
@@ -27,5 +29,13 @@ public record Options(
...
               defaultValue = "false")
-          boolean upsert) {}
+          boolean upsert,
+      @Nullable
+          @Schema(
+              description = "Next page state for pagination.",
+              type = SchemaType.STRING,
+              implementation = String.class)
+          @JsonProperty("pageState")
+          @JsonAlias("pagingState") // old name, 1.0.0-BETA-3 and prior
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/UpdateManyCommand.java:39]: Do we need this also here? I know it is used for older cases. I guess that makes sense for consistency, and we need not document it (i.e. only include main new option in descriptions).

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/UpdateManyIntegrationTest.java [hunk 2]
```diff
@@ -284,6 +289,80 @@ public void limitMoreDataFlag() {
...
+              .body("status.matchedCount", is(20))
+              .body("status.modifiedCount", is(20))
+              .body("status.moreData", is(true))
+              .body("status.nextPageState", notNullValue())
+              .body("errors", is(nullValue()))
+              .extract()
+              .body()
+              .path("status.nextPageState");
+
+      System.out.println("nextPageState : " + nextPageState);
```
Comment (Yuqi-Du) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/UpdateManyIntegrationTest.java:322]: Could we delete this print.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/UpdateOperationPage.java [hunk 3]
```diff
@@ -61,7 +61,10 @@ public CommandResult get() {
     updateStatus.put(CommandStatus.MATCHED_COUNT, matchedCount());
     updateStatus.put(CommandStatus.MODIFIED_COUNT, modifiedCount());
 
-    if (moreDataFlag) updateStatus.put(CommandStatus.MORE_DATA, moreDataFlag);
+    if (pagingState != null) {
+      updateStatus.put(CommandStatus.MORE_DATA, true);
```
Comment (Yuqi-Du) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/UpdateOperationPage.java:65]: This is neat! for backwards compatibility.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/UpdateManyIntegrationTest.java [hunk 4]
```diff
@@ -217,7 +220,8 @@ public void limit() {
           .statusCode(200)
           .body("status.matchedCount", is(20))
           .body("status.modifiedCount", is(20))
-          .body("status.moreData", nullValue())
+          .body("status.moreData", is(true))
+          .body("status.nextPageState", notNullValue())
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/UpdateManyIntegrationTest.java:224]: I know we can't compare to specific value (as it's generated token), but maybe check it's a non-empty String (length above 0)?

**Review by tatu-at-datastax — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/UpdateManyCommand.java (hunk 1 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/UpdateManyCommand.java:39]: Just used what's available in `find` command.

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/UpdateManyCommand.java:39]: Right, figured. That's fine.

**Review by vkarpov15 — APPROVED** [review]
LGTM

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java [hunk 5]
```diff
@@ -55,46 +54,26 @@ public record ReadAndUpdateOperation(
...
-                if (matchedCount.get() + docs.size() <= updateLimit) {
-                  matchedCount.addAndGet(docs.size());
-                  return Multi.createFrom().items(docs.stream());
-                } else {
-                  int needed = updateLimit - matchedCount.get();
-                  matchedCount.addAndGet(needed);
-                  moreDataFlag.set(true);
-                  return Multi.createFrom().items(findResponse.docs().subList(0, needed).stream());
-                }
+                matchedCount.addAndGet(docs.size());
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java:71]: I noticed while working on stargate/stargate-mongoose#199 that this is a breaking change. In v1.0.1, `moreData` would be false if exactly 20 documents were updated. But now `moreData` is true if exactly 20 documents are updated.

Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java:71]: Cassandra returns a page state if page size of records retrieved are retrieved . Reading with this pageState won't not return any more data. Yes, we would need to deprecate `moreData` flag. The looping should be based on `pageState != null`

**Pull request merged**